.svglintrc.js

```diff
@@ -129,6 +131,34 @@ module.exports = {
               }
             }
           },
+          function(reporter, $, ast) {
+            reporter.name = "icon-precision";
+
+            const iconPath = $.find("path").attr("d");
+            const { segments } = parsePath(iconPath);
+            const segmentParts = segments.flat().filter((num) => (typeof num === 'number'));
+
+            const countDecimals = (num) => {
+              if (num && num % 1) {
+                let [base, op, trail] = num.toExponential().split(/e([+-])/);
+                let elen = parseInt(trail, 10);
+                let idx = base.indexOf('.');
+                return idx == -1 ? elen : base.length - idx - 1 + (op === '+' ? -elen : elen);
+              }
+              return 0;
+            };
+            const precisionArray = segmentParts.map(countDecimals);
+            const precisionMax = precisionArray && precisionArray.length > 0 ?
+              Math.max(...precisionArray) :
+              0;
+
+            if (precisionMax > iconMaxFloatPrecision) {
+              reporter.error(`Precision level should not be greater than ${iconMaxFloatPrecision}; the maximum is currently ${precisionMax}`);
```

Comment:
I think this should be reworded for clarity, what about:

```suggestion
              reporter.error(`Precision level should not be greater than ${iconMaxFloatPrecision}; the precision is currently ${precisionMax}`);
```

Reply:
I suggest:

```suggestion
              reporter.error(`Maximum precision level should not be greater than ${iconMaxFloatPrecision}; the precision is currently ${precisionMax}`);
```

Reply:
I personally see "maximum precision" and "precision" (in this context) as the same thing, but I don't mind being explicit about it. The only thing I worry about is that the message is getting pretty long... How about:

- Removing "level" from the first sentence?
- Changing the second sentence to `it is currently ${precisionMax}`?

```suggestion
              reporter.error(`Maximum precision should not be greater than ${iconMaxFloatPrecision}; it is currently ${precisionMax}`);
```

Reply:
> - Removing "level" from the first sentence?
> - Changing the second sentence to it is currently ${precisionMax}?

Totally agree.

Reply:
English is not my native language as you can see 😛